pkg/health/condition/check_backup_ready.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package condition
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	coordinationv1 "k8s.io/api/coordination/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type backupReadyCheck struct {
+	cl client.Client
+}
+
+const (
+	BackupSucceeded string = "BackupSucceeded"
+	BackupFailed    string = "BackupFailed"
+	Unknown         string = "Unknown"
+)
+
+func (a *backupReadyCheck) Check(ctx context.Context, etcd druidv1alpha1.Etcd) Result {
+	//Default case
+	result := &result{
+		conType: druidv1alpha1.ConditionTypeBackupReady,
+		status:  druidv1alpha1.ConditionUnknown,
+		reason:  Unknown,
+		message: "Cannot determine etcd backup status",
+	}
+
+	//Fetch snapshot leases
+	fullSnapLease := &coordinationv1.Lease{}
+	deltaSnapLease := &coordinationv1.Lease{}
+	var fullSnapErr, incrSnapErr error
+	fullSnapErr = a.cl.Get(ctx, types.NamespacedName{Name: getFullSnapLeaseName(&etcd), Namespace: etcd.ObjectMeta.Namespace}, fullSnapLease)
+	incrSnapErr = a.cl.Get(ctx, types.NamespacedName{Name: getDeltaSnapLeaseName(&etcd), Namespace: etcd.ObjectMeta.Namespace}, deltaSnapLease)
+
+	//Set status to Unknown if errors in fetching snapshot leases or lease never renewed
+	if fullSnapErr != nil || incrSnapErr != nil || (fullSnapLease.Spec.RenewTime == nil && deltaSnapLease.Spec.RenewTime == nil) {
+		return result
+	}
+
+	deltaLeaseRenewTime := deltaSnapLease.Spec.RenewTime
+	fullLeaseRenewTime := fullSnapLease.Spec.RenewTime
+	fullLeaseCreateTime := &fullSnapLease.ObjectMeta.CreationTimestamp
+
+	if fullLeaseRenewTime == nil && deltaLeaseRenewTime != nil && fullLeaseCreateTime != nil {
```

Comment:
Above there is:
```
fullLeaseCreateTime := &fullSnapLease.ObjectMeta.CreationTimestamp
```
So `fullLeaseCreateTime` will never be `nil`. Can you double check?

Reply:
Snapshot leases contain `CreationTimestamp` when they're created and any updates to the snapshot leases happen through patches
That should be sufficient to guarantee that `CreationTimestamp` always exists, right?

Reply:
I think you don't need to check this because in line 57 you retrieve the full snapshot lease, then in line 61 you check for errors and if there isn't one then you certainly got the object from the Apiserver (no not found error) and thus it'll always have a `CreationTimestamp`.

Reply:
Ah, okay
So `fullLeaseCreateTime != nil` is kind of redundant in this condition
did I understand correctly what you're trying to imply?

Reply:
Updated condition